packages/admin/admin/src/mui/MasterLayout.tsx

```diff
@@ -65,6 +72,24 @@ export function MasterLayout(inProps: MasterLayoutProps) {
 
     const [open, setOpen] = useState(openMenuByDefault);
     const [drawerVariant, setDrawerVariant] = useState<"permanent" | "temporary">("permanent");
+    const [menuWidth, setMenuWidth] = useState(0);
+    const menuRef = useRef<HTMLDivElement>(null);
+
+    useEffect(() => {
+        if (!menuRef.current) return;
+
+        const resizeObserver = new ResizeObserver((entries) => {
+            for (const entry of entries) {
+                setMenuWidth(entry.contentRect.width);
+            }
+        });
```

Comment:
```suggestion
        const resizeObserver = new ResizeObserver(([entry]) => {
            setMenuWidth(entry.contentRect.width);
        });
```

IMO this would be less confusing.

Reply:
True, but I added an `if` for safety: b73cd09644eba203972ba219f8aea2cbacfbec4f